### PR TITLE
Update content about upgrade

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6474
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: upgrade-tidb-using-tiup.md [hunk 1]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
...
 - To update versions earlier than 3.0 to 5.0:
     1. Update this version to 3.0 using [TiDB Ansible](https://docs.pingcap.com/tidb/v3.0/upgrade-tidb-using-ansible).
     2. Use TiUP (`tiup cluster`) to import the TiDB Ansible configuration.
     3. Update the 3.0 version to 4.0 according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup).
-    4. Then you can upgrade the cluster to v5.0 according to this document.
+    4. Then you can upgrade the cluster to v5.1 according to this document.
 - Support upgrading the versions of TiDB Binlog, TiCDC, TiFlash, and other components.
+- For detailed compatibility changes of different versions, see the [Release Notes](/releases/release-notes.md) of each version. Modify your cluster configuration according the "Compatibility Changes" section of the corresponding release notes.
```
Comment (TomShawn) [inline upgrade-tidb-using-tiup.md:28]: @lilyjazz 
```suggestion
- For detailed compatibility changes of different versions, see the [Release Notes](/releases/release-notes.md) of each version. Modify your cluster configuration according to the "Compatibility Changes" section of the corresponding release notes. If the release notes of a version do not have this section, you do not need to make compatibility changes in your cluster configuration.
```

File: upgrade-tidb-using-tiup.md [hunk 2]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
 
 # Upgrade TiDB Using TiUP
 
-This document is targeted for users who want to upgrade from TiDB 4.0 versions to TiDB 5.0 versions, or from TiDB 5.0 to a later version.
+This document is targeted for the following upgrade paths:
 
-If your cluster to upgrade is v3.1 or an earlier version (v3.0 or v2.1), you need to upgrade your cluster first to v4.0 and then to v5.0. You cannot upgrade your cluster across a major version or versions.
+- Upgrade from TiDB 4.0 versions to TiDB 5.1 versions, or from TiDB 5.1 to a later v5.1 patch version.
+- Upgrade from TiDB 5.0 versions to TiDB 5.1 versions, or from TiDB 5.1 to a later v5.1 patch version.
```
Comment (qiancai) [inline upgrade-tidb-using-tiup.md:12]:
```suggestion
- Upgrade from TiDB 4.0 versions to TiDB 5.1 versions.
- Upgrade from TiDB 5.0 versions to TiDB 5.1 versions,
```

Comment (qiancai) [reply upgrade-tidb-using-tiup.md:12]: perhaps, we could mention " Upgrade TiDB 5.1 to a later v5.1 patch version. " as the 3rd item after we have the first 5.1 patch version?

File: upgrade-tidb-using-tiup.md [hunk 3]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
...
+
+> **Note:**
+>
+> If your cluster to upgrade is v3.1 or an earlier version (v3.0 or v2.1), direct upgrade to v5.1 or its patch versions is not supported. You need to upgrade your cluster first to v4.0 and then to v5.1.
```
Comment (qiancai) [inline upgrade-tidb-using-tiup.md:16]:
```suggestion
> If your cluster to be upgraded is v3.1 or an earlier version (v3.0 or v2.1), the direct upgrade to v5.1 or its patch versions is not supported. You need to upgrade your cluster first to v4.0 and then to v5.1.
```

File: upgrade-tidb-using-tiup.md [hunk 4]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
...
+- TiDB currently does not support version downgrade or rolling back to an earlier version after the upgrade.
+- For the v4.0 cluster managed using TiDB Ansible, you need to import the cluster to TiUP (`tiup cluster`) for new management according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup). Then you can upgrade the cluster to v5.1 or its patch versions according to this document.
 - To update versions earlier than 3.0 to 5.0:
     1. Update this version to 3.0 using [TiDB Ansible](https://docs.pingcap.com/tidb/v3.0/upgrade-tidb-using-ansible).
     2. Use TiUP (`tiup cluster`) to import the TiDB Ansible configuration.
     3. Update the 3.0 version to 4.0 according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup).
-    4. Then you can upgrade the cluster to v5.0 according to this document.
+    4. Then you can upgrade the cluster to v5.1 according to this document.
```
Comment (qiancai) [inline upgrade-tidb-using-tiup.md:26]:
```suggestion
    4. Upgrade the cluster to v5.1 according to this document.
```

File: upgrade-tidb-using-tiup.md [hunk 5]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
...
 
 ## Upgrade caveat
 
-- After the upgrade, rolling back to 4.0 or earlier versions is not supported.
-- For the v4.0 cluster managed using TiDB Ansible, you need to import the cluster to TiUP (`tiup cluster`) for new management according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup). Then you can upgrade the cluster to v5.0 according to this document.
+- TiDB currently does not support version downgrade or rolling back to an earlier version after the upgrade.
+- For the v4.0 cluster managed using TiDB Ansible, you need to import the cluster to TiUP (`tiup cluster`) for new management according to [Upgrade TiDB Using TiUP (v4.0)](https://docs.pingcap.com/tidb/v4.0/upgrade-tidb-using-tiup#import-tidb-ansible-and-the-inventoryini-configuration-to-tiup). Then you can upgrade the cluster to v5.1 or its patch versions according to this document.
 - To update versions earlier than 3.0 to 5.0:
```
Comment (qiancai) [inline upgrade-tidb-using-tiup.md:22]:
```suggestion
- To update versions earlier than 3.0 to 5.1:
```

File: upgrade-tidb-using-tiup.md [hunk 6]
```diff
@@ -6,20 +6,26 @@ aliases: ['/docs/dev/upgrade-tidb-using-tiup/','/docs/dev/how-to/upgrade/using-t
 
 # Upgrade TiDB Using TiUP
 
-This document is targeted for users who want to upgrade from TiDB 4.0 versions to TiDB 5.0 versions, or from TiDB 5.0 to a later version.
+This document is targeted for the following upgrade paths:
 
-If your cluster to upgrade is v3.1 or an earlier version (v3.0 or v2.1), you need to upgrade your cluster first to v4.0 and then to v5.0. You cannot upgrade your cluster across a major version or versions.
+- Upgrade from TiDB 4.0 versions to TiDB 5.1 versions.
+- Upgrade from TiDB 5.0 versions to TiDB 5.1 versions,
```
Comment (qiancai) [inline upgrade-tidb-using-tiup.md:12]:
```suggestion
- Upgrade from TiDB 5.0 versions to TiDB 5.1 versions.
```

**Review by qiancai — APPROVED** [review]

**Review by lilyjazz — APPROVED** [review]

**Pull request merged**